vaas/vaas/adminext/widgets.py

```diff
@@ -157,14 +157,14 @@ class ComplexRedirectConditionField(forms.MultiValueField):
     def __init__(self, **kwargs):
         configuration = provide_redirect_configuration() 
         http_methods = tuple((http_method.http_method, http_method.name) for http_method in configuration.http_methods)
-        domains = tuple((domain.domain, domain.domain) for domain in configuration.domains)
+        domains = tuple((domain.domain, domain.pk) for domain in configuration.domains)
```

Comment:
We use this pk before saving model in to database and gather DomainMapping object by this pk.

Reply:
ok, but by the book usage of choice is:

```
forms.ChoiceField(choices=((1, 'label for one'), (2, 'label for two'), ...))
```
in your case the order is reversed:

```
forms.ChoiceField(choices=(('label for one', 1), ('label for two', 2), ...))
```

it could be a little misleading.

Reply:
Ok lets change this order for non-reversed:
![Screenshot 2022-12-02 at 07 39 24](https://user-images.githubusercontent.com/4668967/205231336-0c5d586e-678e-4cfc-85b1-b1b8a66971bf.png)
As we see on screen above widget print the first value. The second value will be save to this field.
We wish show name instead of id.
I can change it for domain.domain, domain.doman and get DomainMapping object not by pk but by name. 
You can decide.

Reply:
I would rather recommend checking why it's not behaving as supposed. Two minutes of digging pointing to mistaken construction of RedirectConfiguration object ... what a surprise?

Reply:
I will save your priceless time: https://github.com/allegro/vaas/pull/432/files#diff-5a35c0dd66732d79592525f5e868c546c573709dc6918a9d7a8deb5ee8250f6cR150

Reply:
I forgot we change there order... I was digging in ModelField _choice method.